narwhals/_arrow/utils.py

```diff
@@ -418,5 +420,12 @@ def _parse_date_format(arr: pa.Array) -> str:
 def _parse_time_format(arr: pa.Array) -> str:
     import pyarrow.compute as pc  # ignore-banned-import
 
-    matches = pc.extract_regex(arr, pattern=TIME_RE)
-    return "%H:%M:%S" if pc.all(matches.is_valid()).as_py() else ""
+    format = ""
+    matches = pc.extract_regex(arr, pattern=HMS_RE)
+    if pc.all(matches.is_valid()).as_py():
+        format = "%H:%M:%S"
+    else:
+        matches = pc.extract_regex(arr, pattern=HM_RE)
+        if pc.all(matches.is_valid()).as_py():
+            format = "%H:%M"
+    return format
```

Comment:
Opinionated take is to follow a similar pattern (pun unintended) as per date regex, i.e. having a format mapping:

```py
TIME_FORMATS = (
    (HMS_RE, "%H:%M:%S"),
    (HM_RE, "%H:%M"),
)
```

and here loop through that and have a early return for the first that fully matches, otherwise return empty string

Reply:
Thank you for the suggestion @FBruzzesi! I added a time formatting mapping similar to how it is done for dates, it looks much nicer now :tada: